includes/AMP/Output_Buffer.php

```diff
@@ -0,0 +1,295 @@
+<?php
+/**
+ * Class Output_Buffer
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\AMP;
+
+use DOMElement;
+use Exception;
+use Google\Web_Stories\Experiments;
+use Google\Web_Stories\Model\Story;
+use Google\Web_Stories\Service_Base;
+use Google\Web_Stories\Settings;
+use Google\Web_Stories\Story_Post_Type;
+use Google\Web_Stories\Traits\Publisher;
+use Google\Web_Stories_Dependencies\AmpProject\Dom\Document;
+use Throwable;
+use WP_Post;
+
+/**
+ * Output buffer class.
+ *
+ * Largely copied from AMP_Theme_Support.
+ *
+ * @since 1.10.0
+ *
+ * @see \AMP_Theme_Support
+ */
+class Output_Buffer extends Service_Base {
+	use Publisher;
```

Comment:
Changing this to a service making it much easier to inject a publisher object into all the places that use this trait 👍

Reply:
We could think about making `Publisher` an actual class, if that's what you mean.

Reply:
> making Publisher an actual class,

Yeah, that is what I meant.